### PR TITLE
labels added, custom cmaps added also if they are not in styles

### LINKED ISSUE #1046: Configure custom color maps in xcube server
**Is your feature request related to a problem? Please describe.**

Since xcube 1.6, xcube server accepts custom color maps from xcube viewer. We want to allow configuring the same type of custom color maps in xcube server, namely

- Continuous
- Stepwise
- Categorical

**Describe the solution you'd like**

```yaml
CustomColorMaps:
  - Identifier: my_categorical_cmap
    Type: categorical     # or continuous, stepwise 
    Colors:
    - Value: 0
      Color: red
      Label: low
    - Value: 1
      Color: #0000FF
      Label: medium
    - Value: 2
      Color: 0,0,255
      Label: high
  - ...
```

EDIT:

An entry in the in the `Colors` list may have one of the following forms (TypeScript notation):

```ts

type ColorEntry = ColorEntryObj | ColorEntryTuple;

interface ColorEntryObj {
  Value: number;
  Color: Color;
  Label?: string;
}

type ColorEntryTuple = [number, Color] | [number, Color: label];

type Color = ColorName | ColorHex | ColorTuple;
type ColorName = <A CSS color name>; 
type ColorHex = <A CSS color in hexdecimal form>;
type ColorTuple = [number, number, number] | [number, number, number, number]
```

### PR DESCRIPTION
**Pull request by konstntokas**
Closes #1046 second part; see [discussion comment and comments below](https://github.com/xcube-dev/xcube/issues/1046#issuecomment-2273583595)

`CHANGES.md` will not be updated, as it is already updated in #1055.  

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [x] New/modified features documented in `docs/source/*`
* [ ] ~Changes documented in `CHANGES.md`~
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Comment by konstntokas** [comment]
- The labels are published now at the endpoint `/colorbar`. 
- All color maps are published given in the section `CustomColorMaps`, even if they are not used in `Styles`

Screenshot from 2024-08-08 15-44-25

File: test/util/test_cmaps.py [hunk 1]
```diff
@@ -403,6 +403,7 @@ def test_create_colormap_from_config_color_entry_object(self):
                 "name": "my_cmap",
                 "type": "continuous",
                 "colors": [[0.0, "red"], [12.0, "#0000FF"], [24.0, [0, 1, 0, 0.3]]],
+                "labels": ["low", "medium", "high"],
```
Comment (forman) [inline test/util/test_cmaps.py:406]: That's not correct - at least not what was specified:

` type ColorEntryTuple = [number, Color] | [number, Color, label];`

See also docs.

File: xcube/util/cmaps.py [hunk 2]
```diff
@@ -640,18 +640,25 @@ def create_colormap_from_config(cmap_config: dict) -> Colormap:
...
     for i, [_, color] in enumerate(colors):
         if isinstance(color, list):
             if any([val > 1 for val in color]):
                 colors[i][1] = list(np.array(color) / 255)
     config_parse = dict(
-        name=cmap_config["Identifier"], type=cmap_config["Type"], colors=colors
+        name=cmap_config["Identifier"],
+        type=cmap_config["Type"],
+        colors=colors,
+        labels=labels,
```
Comment (forman) [inline xcube/util/cmaps.py:661]: See comment above.

**Review by forman — CHANGES_REQUESTED** [review]

**Review by forman — APPROVED** [review]
Nice!

**Pull request merged**